map.go

```diff
@@ -964,13 +964,27 @@ func (m *Map) guessNonExistentKey() ([]byte, error) {
 // "keysOut" and "valuesOut" must be of type slice, a pointer
 // to a slice or buffer will not work.
 // "prevKey" is the key to start the batch lookup from, it will
-// *not* be included in the results. Use nil to start at the first key.
+// *not* be included in the results. It is an opaque handle. Use
+// nil to start at the first key. For subsequent lookups,
+// "nextKeyOut" should be passed as "prevKey" to continue the
+// batch operation.
 //
 // ErrKeyNotExist is returned when the batch lookup has reached
 // the end of all possible results, even when partial results
 // are returned. It should be used to evaluate when lookup is "done".
-func (m *Map) BatchLookup(prevKey, nextKeyOut, keysOut, valuesOut interface{}, opts *BatchOptions) (int, error) {
-	return m.batchLookup(sys.BPF_MAP_LOOKUP_BATCH, prevKey, nextKeyOut, keysOut, valuesOut, opts)
+func (m *Map) BatchLookup(cursor *BatchCursor, keysOut, valuesOut interface{}, opts *BatchOptions) (int, error) {
+	if err := m.validateBatchCursor(cursor); err != nil {
+		return 0, err
+	} else if cursor.opaque == nil {
```

Comment:
How about pushing this down into `m.batchLookup` instead? No need to duplicate the code.